docs/upgrade.rst

```diff
@@ -28,4 +30,48 @@ Notes
 1. We do **not** recommend running ``helm upgrade --reuse-values`` `as it doesn't update default values changed in the chart.
 <https://medium.com/@kcatstack/understand-helm-upgrade-flags-reset-values-reuse-values-6e58ac8f127e>`_
 
-2. To install a Robusta pre-release, run ``helm upgrade`` with the ``--devel`` flag.
\ No newline at end of file
+2. To install a Robusta pre-release, run ``helm upgrade`` with the ``--devel`` flag.
+
+Upgrading to a new major version
+------------------------------
+
+If you didnt install Robusta's bundled Prometheus Stack then you can :ref:`Upgrade at ease <Helm Upgrade>`, Otherwise, keep reading.
+
+1. Why do i need to manually upgrade?
+
+Robusta uses Prometheus Stack as a dependency, Prometheus creates Custom resources also known as `CRDs <https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/>`_ on installation.     
+With Helm v3, CRDs are not updated or removed by default and should be manually handled. Consult also the `Helm Documentation on CRDs <https://helm.sh/docs/chart_best_practices/custom_resource_definitions/>`_. 
+
+From 0.8.x to >= 0.9.x 
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+1. Check robusta version:
+
+.. code-block:: bash
+
+    robusta version
+
+2. We suggest removing and installing for the smoothest experience, start by uninstalling Robusta:
+
+.. code-block:: bash
+
+    helm uninstall robusta
+
+3. Manually remove the instlled CRDs (`for more info read here <https://github.com/prometheus-community/helm-charts/tree/main/charts/kube-prometheus-stack#uninstall-chart>`_):
+
+.. code-block:: bash
+
+    kubectl delete crd alertmanagerconfigs.monitoring.coreos.com
```

Comment:
I think there is an edge case here that could be dangerous for users:

1. You have an existing Prometheus Operator running in your cluster
2. You install Robusta with/without the Promtheus feature. (If you install it with, you now have two Prometheus instances running in your cluster as part of different Helm releases. This should work in principal and might be unusual but it's possible.)
3. Someone upgrades Robusta and follows these instructions.
4. Because CRDs are global, this breaks their existing Prometheus install which expects older CRDs.

A few thoughts about solutions:
1. If we can do an upgrade for the CRDs without deleting them first, that's probably safer
2. We should probably warn the user to think this through if they have a separate existing Prometheus install. E.g. put a warning on this whole section like "If you have an existing Prometheus Operator installed independently of Robusta then be very careful! Upgrading CRDs will impact all Prometheus Operators in your cluster."

Reply:
I updated the upgrading guide. i think the with/without prometheus is probably better.
added the info about the edge cases. tried them a bit, and installation seems to fail in these cases. I would say its a very rare edge case

Reply:
Thanks, looks good.